src/components/PostsContainer/PostsContainer.tsx

```diff
@@ -7,19 +7,39 @@ import { Search } from '../Search/Search';
 import { useInputValue } from '../../hooks/useInputValue';
 import { makeGoogleSearchQuery } from '../../utils/makeGoogleSearchQuery';
 import { useSearchParams } from '../../hooks/useSearchParams';
-//todo import { NavigationButtons } from '../NavigationButtons/NavigationButtons';
+import { NavigationButtons } from '../NavigationButtons/NavigationButtons';
 
 export const PostsContainer: React.FC = ({}) => {
   const search = useSearchParams();
   const { input, handleInput } = useInputValue();
+  const memorizedFilter = React.useMemo(
+    () => makeGoogleSearchQuery(search),
+    [search],
+  );
+
+  const { messages, pageToken, getNextMessagesList, getPreviousMessagesList } =
+    useMessages(memorizedFilter);
 
-  const messages = useMessages(makeGoogleSearchQuery(search));
+  const goForward = () => {
```

Comment:
`useCallback`

Reply:
I've done it like this after our yesterday's standup:
`  const memorizedFilter = React.useMemo(
    () => makeGoogleSearchQuery(search),
    [search],
  );
`

is it okay?